### PR TITLE
Doctrine filters: Allow nested properties (across relations)

### PR DESCRIPTION
**Pull request by vincentchalamon**
Same PR as https://github.com/dunglas/DunglasApiBundle/pull/242, but this one is for 1.0

### CODE REVIEW

**Comment by vincentchalamon** [comment]
@teohhanhui Could you review it to ensure I didn't forget anything and did not make any mistake with cherry-picking [our PR](https://github.com/dunglas/DunglasApiBundle/pull/242) ?

**Comment by MLKiiwy** [comment]
:+1:  :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.0.0 beta 4
```
Comment (dunglas) [inline CHANGELOG.md]: 1.1

**Comment by teohhanhui** [comment]
:+1

But I imagine someone might get bitten by the Doctrine paginator. Perhaps worth adding a note on how to properly configure your custom paginator to work with this (with the help of `QueryChecker`)...

**Comment by teohhanhui** [comment]
https://github.com/dunglas/DunglasApiBundle/blob/1.0/Doctrine/Orm/DataProvider.php#L153-L154

Yes, it should be done.

**Comment by teohhanhui** [comment]
On a related note, I never really answered that comment on performance. Someone might want to benchmark that if it matters...

**Comment by vincentchalamon** [comment]
About Doctrine paginator, I think it should be done in another PR

**Comment by teohhanhui** [comment]
Isn't the support for nested properties a little useless if it throws up so easily (when not using output walkers)? That's why I had to implement all those checks in `QueryChecker`...

**Comment by teohhanhui** [comment]
I mean, shouldn't we just do the same thing like we did in #242?

```php
$doctrineOrmPaginator->setUseOutputWalkers($this->useOutputWalkers($queryBuilder));
```

And considering that we already have access to the Doctrine `ManagerRegistry` in `DataProvider`, it's doesn't even require any other changes.

**Comment by vincentchalamon** [comment]
Following code is done in PaginationExtension, this class does not exist in 1.0:
```php
$doctrineOrmPaginator->setUseOutputWalkers($this->useOutputWalkers($queryBuilder));
```
If it has to be created, this work should be done in a separate PR isn't it ?

**Comment by teohhanhui** [comment]
@vincentchalamon See https://github.com/dunglas/DunglasApiBundle/blob/1.0/Doctrine/Orm/DataProvider.php#L153-L154

It was just refactored...

**Comment by vincentchalamon** [comment]
You're right. I transpose immediately `useOutputWalkers` method on `DataProvider` class

**Comment by dunglas** [comment]
@vincentchalamon @teohhanhui Can I merge this PR? I'll release 1.1 beta soon.

**Comment by teohhanhui** [comment]
LGTM

**Comment by dunglas** [comment]
Thank you @teohhanhui @vincentchalamon

**Comment by dunglas** [comment]
Can you update the doc (https://github.com/api-platform/doc) to mention this new feature?

**Comment by vincentchalamon** [comment]
Thank you @teohhanhui for this great work :)

**Pull request merged**